internal/service/studies/assets.go

```diff
@@ -0,0 +1,191 @@
+	}
+
+	// Validate locations
+	if len(assetData.Locations) == 0 {
+		return &openapi.AssetCreateValidationError{ErrorMessage: "at least one location must be specified"}, nil
+	}
+
+	// Validate protection field
+	if strings.TrimSpace(string(assetData.Protection)) == "" {
+		return &openapi.AssetCreateValidationError{ErrorMessage: "protection is required"}, nil
+	}
+	validProtections := []openapi.AssetBaseProtection{
+		openapi.AssetBaseProtectionAnonymisation,
+		openapi.AssetBaseProtectionPseudonymisation,
+		openapi.AssetBaseProtectionIdentifiableLowConfidencePseudonymisation,
+	}
+	if !slices.Contains(validProtections, assetData.Protection) {
+		return &openapi.AssetCreateValidationError{ErrorMessage: "protection must be one of: anonymisation, pseudonymisation, identifiable_low_confidence_pseudonymisation"}, nil
+	}
+
+	// Validate legal_basis
+	if strings.TrimSpace(assetData.LegalBasis) == "" {
+		return &openapi.AssetCreateValidationError{ErrorMessage: "legal_basis is required"}, nil
+	}
+
+	// Validate format field
+	if strings.TrimSpace(string(assetData.Format)) == "" {
+		return &openapi.AssetCreateValidationError{ErrorMessage: "format is required"}, nil
+	}
+	validFormats := []openapi.AssetBaseFormat{
+		openapi.AssetBaseFormatElectronic,
+		openapi.AssetBaseFormatPaper,
+		openapi.AssetBaseFormatOther,
+	}
+	if !slices.Contains(validFormats, assetData.Format) {
+		return &openapi.AssetCreateValidationError{ErrorMessage: "format must be one of: electronic, paper, other"}, nil
+	}
+
+	// Validate expiry field
+	if strings.TrimSpace(assetData.Expiry) == "" {
+		return &openapi.AssetCreateValidationError{ErrorMessage: "expiry date is required"}, nil
+	}
+
+	// Validate status field
+	if strings.TrimSpace(string(assetData.Status)) == "" {
+		return &openapi.AssetCreateValidationError{ErrorMessage: "status is required"}, nil
+	}
+	validStatuses := []openapi.AssetBaseStatus{
+		openapi.AssetBaseStatusActive,
+		openapi.AssetBaseStatusAwaiting,
+		openapi.AssetBaseStatusDestroyed,
+	}
+	if !slices.Contains(validStatuses, assetData.Status) {
+		return &openapi.AssetCreateValidationError{ErrorMessage: "status must be one of: active, awaiting, destroyed"}, nil
+	}
+
+	// Validate third_party_agreement when accessed_by_third_parties is true
+	if assetData.AccessedByThirdParties && strings.TrimSpace(assetData.ThirdPartyAgreement) == "" {
+		return &openapi.AssetCreateValidationError{ErrorMessage: "third_party_agreement is required when accessed_by_third_parties is true"}, nil
+	}
+
+	// Check if the Asset title already exists
+	// var count int64
+	// err := s.db.Model(&types.Asset{}).Where("title = ?", assetData.Title).Count(&count).Error
+	// if err != nil {
+	// 	return nil, types.NewErrServerError(fmt.Errorf("failed to check for duplicate asset title: %w", err))
+	// }
+	// if count > 0 {
+	// 	return &openapi.AssetCreateValidationError{ErrorMessage: fmt.Sprintf("an asset with the title [%v] already exists", assetData.Title)}, nil
+	// }
```

Comment:
I'm guessing duplicate asset titles are ok? If so I'll delete this bit.

Reply:
I think so